### PR TITLE
add list, run cli flags to integration tests.

### PR DESCRIPTION
**Pull request by abhijeetbhagat**
fixes: #338

### CODE REVIEW

File: test/integration-test/src/main.rs [hunk 1]
```diff
@@ -3,14 +3,59 @@ use log::info;
...
-            panic!("{}", e)
+    let cmd = Command::parse();
+
+    match cmd {
+        Command::Run(opts) => {
+            match opts.tests {
+                Some(tests) => {
+                    for t in inventory::iter::<IntegrationTest> {
+                        if tests.contains(&t.name.into()) {
+                            info!("Running {}", t.name);
```
Comment (vadorovsky) [inline test/integration-test/src/main.rs:38]: nit: Maybe let's move

```rust
info!("Running {}", t.name);
if let Err(e) = (t.test_fn)() {
    panic!("{}", e)
}
```

to some common function/macro, to not repeat that twice.

**Comment by vadorovsky** [comment]
Lint check fails.

**Comment by abhijeetbhagat** [comment]
running `cargo clippy --workspace --exclude integration-test -- --deny warnings` on my system doesn't give any warnings.

**Comment by vadorovsky** [comment]
That's because CI is running with Rust nightly. You are going to see those errors when you do:

```
cargo +nightly clippy --workspace --exclude integration-test -- --deny warnings
```

**Comment by abhijeetbhagat** [comment]
nothing:
```
(rust:nightly) ↪ cargo +nightly clippy --workspace --exclude integration-test -- --deny warnings
   Compiling aya-bpf-cty v0.2.1 (/home/abhi/code/aya/bpf/aya-bpf-cty)
   Compiling aya-bpf-bindings v0.1.0 (/home/abhi/code/aya/bpf/aya-bpf-bindings)
   Compiling aya-bpf v0.1.0 (/home/abhi/code/aya/bpf/aya-bpf)
   Compiling aya-bpf-macros v0.1.0 (/home/abhi/code/aya/aya-bpf-macros)
   Compiling aya v0.11.0 (/home/abhi/code/aya/aya)
    Checking aya-tool v0.1.0 (/home/abhi/code/aya/aya-tool)
    Checking xtask v0.1.0 (/home/abhi/code/aya/xtask)
    Checking integration-test-macros v0.1.0 (/home/abhi/code/aya/test/integration-test-macros)
    Checking integration-ebpf v0.1.0 (/home/abhi/code/aya/test/integration-ebpf)
    Checking aya-log-common v0.1.12-dev.0 (/home/abhi/code/aya/aya-log-common)
    Checking aya-log v0.1.12-dev.0 (/home/abhi/code/aya/aya-log)
    Checking aya-log-parser v0.1.11-dev.0 (/home/abhi/code/aya/aya-log-parser)
    Checking aya-log-ebpf-macros v0.1.0 (/home/abhi/code/aya/aya-log-ebpf-macros)
    Checking aya-log-ebpf v0.1.0 (/home/abhi/code/aya/bpf/aya-log-ebpf)
    Finished dev [unoptimized + debuginfo] target(s) in 6.31s
abhi at hyperion in ~/c/aya on integration-tests-cli-options [✓]
(rust:nightly) ↪ rustc --version
rustc 1.66.0-nightly (f83e0266c 2022-10-03)
```

**Comment by vadorovsky** [comment]
Try `rustup update`, there is a newer version of nightly available (you have `f83e0266c 2022-10-03`, the newest one is `3cdf1ab25 2022-10-07`).

**Comment by abhijeetbhagat** [comment]
ok, i get those warnings now. should i fix them all in the same PR?

**Comment by vadorovsky** [comment]
Maybe do a separate one, since those warnings are not related to your changes and they appear on the current `main` branch too.

File: test/integration-test/src/main.rs [hunk 2]
```diff
@@ -3,14 +3,60 @@ use log::info;
...
+#[derive(Debug, Parser)]
+pub struct RunOptions {
+    #[clap(short, long, value_parser)]
+    tests: Option<Vec<String>>,
+}
+
+#[derive(Debug, Parser)]
+struct Options {
+    #[clap(subcommand)]
+    command: Command,
```
Comment (dave-tucker) [inline test/integration-test/src/main.rs:16]:
```rust
    command: Option<Command>,
```

File: test/integration-test/src/main.rs [hunk 3]
```diff
@@ -3,14 +3,60 @@ use log::info;
...
 
+use clap::Parser;
+#[derive(Debug, Parser)]
+pub struct RunOptions {
+    #[clap(short, long, value_parser)]
+    tests: Option<Vec<String>>,
+}
+
+#[derive(Debug, Parser)]
+struct Options {
```
Comment (dave-tucker) [inline test/integration-test/src/main.rs:14]:
```rust
struct Cli {
```

File: test/integration-test/src/main.rs [hunk 4]
```diff
@@ -3,14 +3,60 @@ use log::info;
...
 use tests::IntegrationTest;
 
+use clap::Parser;
+#[derive(Debug, Parser)]
+pub struct RunOptions {
+    #[clap(short, long, value_parser)]
+    tests: Option<Vec<String>>,
+}
+
+#[derive(Debug, Parser)]
```
Comment (dave-tucker) [inline test/integration-test/src/main.rs:13]:
```rust
#[clap(author, version, about, long_about = None)]
#[clap(propagate_version = true)]
```

File: test/integration-test/src/main.rs [hunk 5]
```diff
@@ -3,14 +3,60 @@ use log::info;
...
 
-    // Run the tests
-    for t in inventory::iter::<IntegrationTest> {
-        info!("Running {}", t.name);
-        if let Err(e) = (t.test_fn)() {
-            panic!("{}", e)
+    let cmd = Command::parse();
+
+    match cmd {
+        Command::Run(opts) => match opts.tests {
```
Comment (dave-tucker) [inline test/integration-test/src/main.rs:42]:
```rust
        Some(Command::Run(opts)) => match opts.tests {
```

File: test/integration-test/src/main.rs [hunk 6]
```diff
@@ -3,14 +3,60 @@ use log::info;
...
+                    }
+                }
+            }
+            None => {
+                for t in inventory::iter::<IntegrationTest> {
+                    exec_test!(t)
+                }
+            }
+        },
+        Command::List => {
```
Comment (dave-tucker) [inline test/integration-test/src/main.rs:56]:
```rust
        Some(Command::List) => {
```

File: test/integration-test/src/main.rs [hunk 7]
```diff
@@ -3,14 +3,60 @@ use log::info;
...
+            for t in inventory::iter::<IntegrationTest> {
+                info!("{}", t.name);
+            }
         }
```
Comment (dave-tucker) [inline test/integration-test/src/main.rs:60]:
```rust
        None => {
            for t in inventory::iter::<IntegrationTest> {
                exec_test!(t)
            }
        }
```

File: test/integration-test/src/main.rs [hunk 8]
```diff
@@ -3,14 +3,60 @@ use log::info;
...
+
 fn main() -> anyhow::Result<()> {
     env_logger::init();
 
-    // Run the tests
-    for t in inventory::iter::<IntegrationTest> {
-        info!("Running {}", t.name);
-        if let Err(e) = (t.test_fn)() {
-            panic!("{}", e)
+    let cmd = Command::parse();
```
Comment (dave-tucker) [inline test/integration-test/src/main.rs:39]:
```rust
Cli.parse()
```

File: test/integration-test/src/main.rs [hunk 9]
```diff
@@ -3,14 +3,60 @@ use log::info;
...
+
+    match cmd {
```
Comment (dave-tucker) [inline test/integration-test/src/main.rs:41]:
```rust
   match &cli.command {
```

**Comment by vadorovsky** [comment]
I merged the lint PR. You can rebase this one and I will merge it once CI passes.

**Review by dave-tucker — CHANGES_REQUESTED** [review]
Added some comments to help preserve the existing behaviour and avoid changes to GitHub Workflows and/or xtask.

You might even be able to simply the command match arms further with something like:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5ca0ad4d05a8d71393a4acfa2c4fc7e9

**Review by vadorovsky — APPROVED** [review]

**Review by dave-tucker — APPROVED** [review]
lgtm thanks @abhijeetbhagat

**Pull request merged**